apps/transformers/lib/transformations/arithmetic_add.ex

```diff
@@ -0,0 +1,61 @@
+defmodule Transformers.ArithmeticAdd do
+  @behaviour Transformation
+
+  alias Transformers.Validations.NotBlank
+  alias Transformers.Validations.ValidationStatus
+
+  @impl Transformation
+
+  @addends "addends"
+  @target_field "targetField"
+
+  def transform(payload, parameters) do
+    with {:ok, [addends, target_field]} <- validate(parameters),
+         {:ok, value} <- sumValues(addends, payload) do
+      modified_payload = Map.put(payload, target_field, value)
+      {:ok, modified_payload}
+    else
+      {:error, reason} -> {:error, reason}
+    end
+  end
+
+  defp sumValues(addends, payload) do
+    Enum.reduce_while(addends, {:ok, 0}, fn addend, {:ok, acc} ->
```

Comment:
thought this reduce_while was pretty slick, but I don't love that we had to use multiple `cond` in here

Reply:
Yeah, I haven't found a concise Elixir counterpart to something like a ternary. You could try to do a one-liner such as `result = if is_number(addend), do: addend, else: Map.get(payload, addend)`. Not sure if that's more or less readable. Perhaps if you did that as part of a map before hand, then passed that in instead of the raw addends to the reduce_while?

Reply:
My logic for doing the map inside of the while was so we only had to loop through the payload once. 

I'm not sure how huge the datasets can get, or if we care about performance to such an extent that we care about things like that.

Reply:
Keeping performance in mind is a good idea. Each message payload can be arbitrarily large, though from what I've seen they're usually not bad. But that's just what we've seen not a guarantee.

I'm good with it either way. If you feel the readability could be better, extracting some well named private functions for each conditional might help. Up to you.

Reply:
we (read: Ian) made it a lot more slick 👍